### PR TITLE
Remove trailing whitespace on grep command in prepare_release.sh

### LINKED ISSUE #8304: Remove trailing whitespace in prepare_release.sh
On 3406564b012e2a1f909f9f9708e821d67133632a (nearly 3.5.0):

```
$ ./scripts/prepare_release.sh
$ grep ' $' programs/.gitignore
# 
```

We shouldn't have this trailing whitespace. `check_files.py` will complain about it.

The goal of this task is that after running `./scripts/prepare_release.sh` and committing the result, `check_files.py` is happy.

### PR DESCRIPTION
**Pull request by mcagriaksoy**
## Description

I removed extra character after the `#` so that the problem solved.
Test:
image

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** Not required this is a release helper script.
- [x] **backport** Not required, the script is not present in 2.28
- [x] **tests** Not required

Resolves #8304

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: scripts/prepare_release.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ for GITIGNORE in $GITIGNORES; do
         sed -i 's/###START_COMMENTED_GENERATED_FILES###/###START_GENERATED_FILES###/' $GITIGNORE
         sed -i 's/###END_COMMENTED_GENERATED_FILES###/###END_GENERATED_FILES###/' $GITIGNORE
     else
-        sed -i '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/# /' $GITIGNORE
+        sed -i '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/#/' $GITIGNORE
```
Comment (tom-cosgrove-arm) [inline scripts/prepare_release.sh:58]: This isn't the correct fix.

The original line was putting a comment + space at the beginning of each line from `START_GENERATED_FILES` to `END_GENERATED_FILES` ready for the `sed` at line 54 to remove later in "unrelease" mode.

By removing this space, line 54 will no longer remove the comment, so this script will be broken

Comment (minosgalanakis) [reply scripts/prepare_release.sh:58]: I would argue that this half the fix. As @tom-cosgrove-arm said the unrelease expression in line 54 needs to be updated as well.

`sed -i '/###START_COMMENTED_GENERATED_FILES###/,/###END_COMMENTED_GENERATED_FILES###/s/^#//' $GITIGNORE`

@mcagriaksoy  could you please update the PR and test it using:

~~~~~
# Prepare a release
./scripts/prepare_release.sh &&  cat programs/.gitignore

# Check that there is no trailing whitespace added
 tests/scripts/check_files.py

# Return to devel mode and make sure the git status is clean
./scripts/prepare_release.sh -u
~~~~~~~~~~

Thanks for your contribution.

Comment (gilles-peskine-arm) [reply scripts/prepare_release.sh:58]: As I've mentioned before, I think we should remove `-u` mode. We don't actually need it. Our process for “return to devel mode” is `git revert`, not running a script.

**Comment by minosgalanakis** [comment]
Closing as issue has been fixed.

**Pull request closed without merging**